include/mbedtls/ssl.h

```diff
@@ -2946,6 +2945,12 @@ void mbedtls_ssl_conf_dhm_min_bitlen( mbedtls_ssl_config *conf,
  * \note           This list should be ordered by decreasing preference
  *                 (preferred curve first).
  *
+ * \note           The default list is the same set of curves that
+ *                 #mbedtls_x509_crt_profile_default allows, plus
+ *                 ECDHE-only curves selected according to the same criteria.
+ *                 Larger (generally more secure but slower) curves are
+ *                 preferred over smaller curves.
```

Comment:
_Should_ we prefer larger curves? I kept what it was before, but I'm not completely sure it's right to prioritize bp512r1 (very slow) and secp521r1 (more prone to side channels) over secp384r1 or even secp256r1. Or, for that matter, to prioritize secp384r1 over secp256r1: is the added security really meaningful?

Reply:
In https://github.com/ARMmbed/mbedtls/issues/4127 we wanted to change the order. Note that the current wording unambiguously states that changing the priority order would not be acceptable in a minor release. I don't feel strongly about this: I think the documentation should clarify what is and what isn't guaranteed, but I'm ok with explicitly stating that the priority order may change.

Reply:
After discussing with @mpg and @yanesca : yes, let's change the order to favor performance. A curve is either in or out; if it's in it's secure enough.